File: test/prime.js
// prime.js

function isprime(n, primes, primes_len) {
    const nsqrt = Math.sqrt(n);
    for(var Li = 0; Li < primes_len; ++Li) {
        if(nsqrt < primes[Li]) { break; }
        if(n % primes[Li] == 0) { return false; }
    }
    return true;
}

var primes = [];
var primes_len = 0;
var value = process.argv[process.argv.length - 1]
for(var Li = 2; Li < value; ++Li) {
    if(isprime(Li, primes, primes_len)) {
        primes.push(Li);
        primes_len += 1;
    }
}

console.log(primes[primes_len - 1]);

File: test/prime.java
// prime.java

import java.util.*;

class prime {
    public static boolean isprime(int n, List<Integer> primes) {
        int nsqrt = (int)Math.sqrt(n);
        for(int Li : primes) {
            if(nsqrt < Li) { break; }
            if(n % Li == 0) { return false; }
        }
        return true;
    }
    public static void main(String args[]) {
        int end = Integer.parseInt(args[0]);
        List<Integer> primes = new ArrayList<Integer>();
        for(int Li = 2; Li < end; ++Li) {
            if(isprime(Li, primes)) {
                primes.add(Li);
            }
        }
        System.out.println(primes.get(primes.size() - 1));
    }
}
